### ISSUE TITLE
user_blocks pages lacks proper navigation

### ISSUE THREAD

**Issue by jidanni** [openstreetmap/openstreetmap-website#2387]
At the bottom of
https://www.openstreetmap.org/user_blocks
there is
```
« Previous Page 1 Next »
```
- No way to know how many pages there are.
- No way to jump to the last page.

There may be five more pages.

There may be five hundred.

The only way for the user to find out is to click from one to the next
to the next.

**Comment by mmd-osm** [comment]
We have a few more navigation options in the meantime:

![image](https://github.com/openstreetmap/openstreetmap-website/assets/5842757/947eef03-6b74-4fcc-948f-9d489638adca)

**Comment by kcne** [comment]
Regarding the navigation limitations on the user blocks page, expanding on @mmd-osm answer, I think the current view could use some enhancements. Displaying the total number of pages or blocks would be a nice touch. For example, showing something like "Displaying 20 of ${totalNumberOfBlocks}" or "Page ${currentPage} of ${totalNumberOfPages}" could provide more context and improve the user experience.

Additionally, adding a search input to query blocks by blocked user or creator could significantly boost usability. This would help users efficiently order and find specific entries.

What are you thoughts on this? Would you be willing to merge changes mentioned if I opened a PRs addressing mentioned issues?
@tomhughes @gravitystorm

**Comment by AntonKhorev** [comment]
> Displaying the total number of pages or blocks would be a nice touch. For example, showing something like "Displaying 20 of ${totalNumberOfBlocks}" or "Page ${currentPage} of ${totalNumberOfPages}" could provide more context and improve the user experience.

Some of these things are not done to avoid using `OFFSET`/`COUNT` in db queries.

> Additionally, adding a search input to query blocks by blocked user or creator could significantly boost usability.

They are linked from profile pages. Profile pages also have active blocks and created blocks counters.

Also see #4734 for this from the original post:

> No way to jump to the last page.

**Comment by AntonKhorev** [comment]
If you want to be compatible with https://github.com/openstreetmap/openstreetmap-website/pull/4734 or similar and have some idea about ${totalNumberOfBlocks} while browsing *blocks by* pages, maybe adding a counter inside the *blocks by* tab would help.

**Comment by kcne** [comment]
Regarding the navigation limitations on the user blocks page, one possible solution would be displaying the count of blocks inside the open tab in braces. This would give users a clear context of how many blocks are available, which can be particularly helpful for moderators and others managing the list.

I've already implemented this locally. Please let me know if this approach aligns with what you're looking for, and if so, I can open a PR with these changes.

Screenshots:
<img width="947" alt="Screenshot 2024-07-24 at 20 23 52" src="https://github.com/user-attachments/assets/91101421-3633-4e9c-bc52-27932a67f6e5">

<img width="962" alt="blocks-by-me" src="https://github.com/user-attachments/assets/569ae9f3-7134-47ff-a565-f4c3f3c52227">

<img width="972" alt="blocks-on" src="https://github.com/user-attachments/assets/c8aeb560-75c8-4501-bfb6-3858ba8c175c">



Thank you!

**Comment by AntonKhorev** [comment]
Why in braces? I'm trying to convert such counters to badges, see https://github.com/openstreetmap/openstreetmap-website/pull/4872

**Comment by kcne** [comment]
I agree that in terms of UI it would look a lot cleaner so I will be happy to refactor according to that and leave screenshots when I do

Thanks for a quick response!

**Comment by kcne** [comment]
@AntonKhorev Quick update, refactored this using badges -> [97e7991](https://github.com/openstreetmap/openstreetmap-website/commit/97e79913824f1898c10fe747d57e2012dfa21c97)
It would be nice to ping on #4872 for review so I can open a new PR for this.

Screenshot:
<img width="1094" alt="Screenshot 2024-07-25 at 13 57 36" src="https://github.com/user-attachments/assets/eec17bb1-f45c-4280-b3dc-719595fdb192">
